compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2576,7 +2576,13 @@ TR::Register *OMR::Power::TreeEvaluator::vloadEvaluator(TR::Node *node, TR::Code
    switch(node->getDataType())
      {
      case TR::VectorInt8:
+   cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P9) ? opcode = TR::InstOpCode::lxvb16x : opcode = TR::InstOpCode::lxvw4x;
+   kind = TR_VRF;
+   break;
      case TR::VectorInt16:
+   cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P9) ? opcode = TR::InstOpCode::lxvh8x : opcode = TR::InstOpCode::lxvw4x;
+   kind = TR_VRF;
+   break;
```

Comment:
Yeah, I thought that might be the case, but I actually haven't seen this cause any functional problems yet. I think because (so far), all the operations I've looked at are done element-wise, and since the `lxvw4x` is always paired with `stxvw4x`, the data always ends up in the right place and order when the operation is complete. Of course, that means that other operations that aren't element-wise might run into some problems here.

Unfortunately, the halfword and byte load/store instructions are only available on P9 and higher, so I'm not sure if there's a better alternative for P8 and lower as of right now.

Reply:
I suspect you are right that the pairing of `lxvw4x` and `stxvw4x` is what hides the problem.

One possible fix is to add additional instructions as part of the evaluator to rearrange the data into the correct order after it has been loaded. You would also need to do something similar before it is stored back. I believe this is also only necessary for little endian because the data is kept in the right order under big endian.

Reply:
As mentioned below, we will deal with the P8 and lower case in a separate issue. So we'll address these concerns there

Reply:
Okay, that sounds good to me.